kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala

```diff
@@ -2148,6 +2148,16 @@ object KyuubiConf {
       .booleanConf
       .createWithDefault(false)
 
+  val ENGINE_USER: OptionalConfigEntry[String] =
```

Comment:
Currently, we use `hive.server2.proxy.user` to keep the behavior compatible with hiveserver2

If you want to add an alternative key for it, please check how we handle `hive.server2.proxy.user` in KyuubiAuthenticationFactory. Client user changing need to go through AuthN

Reply:
@yaooqinn Hi, Thanks for the review! So I really forgot to do the proxy validation, because it's essentially a proxy.

Reply:
We can not add a configuration to the switch user outside the control of FrontendServices/Authentication. This varies the basic rule of how we achieve multi-tenancy architecture.

We need a configuration to be an alternative to `hive.server2.proxy.user`. `hive.server2.proxy.user` is derived from the hive server, and it's perfect for keeping the behavior the same as the hive. But hive thrift protocol is only one of the FrontendServices. There are also other FrontendServices not necessary to follow  `hive.server2.proxy.user`. So we need `kyuubi.proxy.user`.

As for #5369, IMHO, it's a misleading task

Reply:
@yaooqinn Ok, I see. I'm going to create another PR to implement the configuration item kyuubi.proxy.user as an option to hive.server2.proxy.user. By the way, when both parameters are provided, which takes precedence? Personally, I think it's kyuubi.proxy.user.

Reply:
Yes, I agree with you.

BTW, for the spark engine with USER and CONNECTION, I'm sure that whether hive.server2.proxy.user is used or not, the client user identity will be used for engine bootstrap and SQL operations. But we need to find out whether other engine types follow this rule.

If you are interested in this topic, you can also have a try.

Reply:
@yaooqinn Ok, I will first add the `kyuubi.proxy.user configuration`. Then I'll confirm how to converge on other engines.

Reply:
Thanks @mrtisttt. I truly appreciate that.